percona-xtradb-cluster-8.0-backup/run_backup.sh

```diff
@@ -11,140 +11,203 @@ LIB_PATH='/usr/lib/pxc'
-    fi
-    vault_store /tmp/${SST_INFO_NAME}
-
-    xbstream -C /tmp -c ${SST_INFO_NAME} \
-        | xbcloud put ${CURL_RET_ERRORS_ARG} ${INSECURE_ARG} --storage=s3 --parallel=10 --md5 --s3-bucket="$S3_BUCKET" "$S3_BUCKET_PATH.$SST_INFO_NAME" 2>&1 \
-        | (grep -v "error: http request failed: Couldn't resolve host name" || exit 1)
-
-    if (( $SST_FAILED == 0 )); then
-         FIRST_RECEIVED=0
-         socat -u "$SOCAT_OPTS" stdio  \
-            | xbcloud put ${CURL_RET_ERRORS_ARG} ${INSECURE_ARG} --storage=s3 --parallel=10 --md5 --s3-bucket="$S3_BUCKET" "$S3_BUCKET_PATH" 2>&1 \
-            | (grep -v "error: http request failed: Couldn't resolve host name" || exit 1)
-         FIRST_RECEIVED=1
-    fi
-
-    mc -C /tmp/mc stat ${INSECURE_ARG} "dest/$S3_BUCKET/$S3_BUCKET_PATH.md5"
-    md5_size=$(mc -C /tmp/mc stat ${INSECURE_ARG} --json "dest/$S3_BUCKET/$S3_BUCKET_PATH.md5" | sed -e 's/.*"size":\([0-9]*\).*/\1/')
-    if [[ $md5_size =~ "Object does not exist" ]] || (($md5_size < 23000)); then
-         log 'ERROR' 'Backup is empty'
-         log 'ERROR' 'Backup was finished unsuccessfull'
-         exit 1
-    fi
+	mc_add_bucket_dest
+
+	socat -u "$SOCAT_OPTS" stdio | xbstream -x -C /tmp &
+	wait $!
+	log 'INFO' 'Socat was started'
+
+	FIRST_RECEIVED=1
+	if [[ $? -ne 0 ]]; then
+		log 'ERROR' 'Socat(1) failed'
+		log 'ERROR' 'Backup was finished unsuccessfully'
+		exit 1
+	fi
+	vault_store /tmp/${SST_INFO_NAME}
+
+	xbstream -C /tmp -c ${SST_INFO_NAME} \
+		| xbcloud put ${CURL_RET_ERRORS_ARG} ${INSECURE_ARG} --storage=s3 --parallel=10 --md5 --s3-bucket="$S3_BUCKET" "$S3_BUCKET_PATH.$SST_INFO_NAME" 2>&1 \
+		| (grep -v "error: http request failed: Couldn't resolve host name" || exit 1)
+
+	if ((SST_FAILED == 0)); then
+		FIRST_RECEIVED=0
+		socat -u "$SOCAT_OPTS" stdio \
+			| xbcloud put ${CURL_RET_ERRORS_ARG} ${INSECURE_ARG} --storage=s3 --parallel=10 --md5 --s3-bucket="$S3_BUCKET" "$S3_BUCKET_PATH" 2>&1 \
+			| (grep -v "error: http request failed: Couldn't resolve host name" || exit 1)
+		FIRST_RECEIVED=1
+	fi
+
+	mc -C /tmp/mc stat ${INSECURE_ARG} "dest/$S3_BUCKET/$S3_BUCKET_PATH.md5"
+	md5_size=$(mc -C /tmp/mc stat ${INSECURE_ARG} --json "dest/$S3_BUCKET/$S3_BUCKET_PATH.md5" | sed -e 's/.*"size":\([0-9]*\).*/\1/')
+	if [[ $md5_size =~ "Object does not exist" ]] || ((md5_size < 23000)); then
+		log 'ERROR' 'Backup is empty'
+		log 'ERROR' 'Backup was finished unsuccessfull'
+		exit 1
+	fi
+}
+
+azure_auth_header() {
+	params="$1"
+	request_date="$2"
+	hex_key="$(echo -n "$AZURE_ACCESS_KEY" | base64 -d -w0 | hexdump -ve '1/1 "%02x"')"
+	headers="x-ms-date:$request_date\nx-ms-version:2021-06-08"
+	resource="/$AZURE_STORAGE_ACCOUNT/$AZURE_CONTAINER_NAME"
+	string_to_sign="GET\n\n\n\n\n\n\n\n\n\n\n\n${headers}\n${resource}\n${params}"
+	signature=$(printf '%s' "$string_to_sign" | openssl dgst -sha256 -mac HMAC -macopt "hexkey:$hex_key" -binary | base64 -w0)
+	echo "Authorization: SharedKey $AZURE_STORAGE_ACCOUNT:$signature"
```

Comment:
I like your bash magic))

Despite this very fact, I have to ask you to switch from manipulations within runtime to temporary files in order
1. make it safer 
2. utilize `curl -H @header_file `

Reply:
https://github.com/percona/percona-docker/pull/623/commits/2c5180eabc563c357aee3f79ae530f8b8762398e